fed/mosaic-application/src/main/java/org/eclipse/mosaic/fed/application/ambassador/simulation/perception/SimplePerceptionModule.java

```diff
@@ -127,9 +133,53 @@ public boolean isInRange(SpatialObject other) {
                 tmpVector2.set(0, 0, 0);
-                return VectorUtils.isLeftOfLine(tmpVector1, tmpVector2, rightBoundVector) // vehicle is left of right edge
-                        && !VectorUtils.isLeftOfLine(tmpVector1, tmpVector2, leftBoundVector) // vehicle is right of left edge
-                        && tmpVector1.magnitude() <= configuration.getViewingRange(); // other vehicle is in range
+
+                if (tmpVector1.magnitude() > configuration.getViewingRange()) { // other vehicle is in range
+                    return false;
+                }
+                if (MathUtils.isFuzzyEqual(configuration.getViewingAngle(), 360d)) { // for 360 degree viewing angle field-of-view check is obsolete
+                    return true;
+                } else if (configuration.getViewingAngle() < 180d) { // for < 180 degree viewing angle we use left and right vector
+                    return isBetweenVectors(tmpVector1, tmpVector2, leftBoundVector, rightBoundVector)
+                            || liesOnVector(tmpVector1, leftBoundVector)
+                            || liesOnVector(tmpVector1, rightBoundVector);
+                } else { // for >= 180 degree do two checks: 1st between direction vector and right or 2nd between direction vector and left
+                    return isBetweenVectors(tmpVector1, tmpVector2, directionVector, rightBoundVector)
+                            || isBetweenVectors(tmpVector1, tmpVector2, leftBoundVector, directionVector)
+                            || liesOnVector(tmpVector1, leftBoundVector)
+                            || liesOnVector(tmpVector1, rightBoundVector)
+                            || liesOnVector(tmpVector1, directionVector);
+                }
+            }
+        }
+
+        private boolean isBetweenVectors(Vector3d pointToEvaluate, Vector3d linePoint, Vector3d leftVector, Vector3d rightVector) {
+            return VectorUtils.isLeftOfLine(pointToEvaluate, linePoint, rightVector) // is left of right edge
+                    && !VectorUtils.isLeftOfLine(pointToEvaluate, linePoint, leftVector); // is right of left vector
+        }
+
+        /**
+         * Calculates the magnitude of the cross product of the two vectors
+         * and checks if they point in the same direction (to avoid objects being located exactly in the opposite direction).
+         * If the magnitude is equal to 0 the point will be on the line, however could be anywhere in each direction, so
+         * we additionally check if the directions match.
+         * Note: This method only works, since we know that we work around the origin.
+         *
+         * @param pointToEvaluate point to be evaluated
+         * @param line            check point against this line
+         * @return {@code true} if point lies on line, otherwise {@code false}
+         */
+        private boolean liesOnVector(Vector3d pointToEvaluate, Vector3d line) {
```

Comment:
This is a simplified version of [this](https://stackoverflow.com/a/11908158) answer.